cli/index_create.go

```diff
@@ -0,0 +1,123 @@
+// Copyright 2023 Democratized Data Foundation
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package cli
+
+import (
+	"bytes"
+	"encoding/json"
+	"io"
+	"net/http"
+	"os"
+
+	"github.com/spf13/cobra"
+
+	httpapi "github.com/sourcenetwork/defradb/api/http"
+	"github.com/sourcenetwork/defradb/client"
+	"github.com/sourcenetwork/defradb/config"
+	"github.com/sourcenetwork/defradb/logging"
+)
+
+type indexCreateResponse struct {
+	Data struct {
+		Index client.IndexDescription `json:"index"`
+	} `json:"data"`
+	Errors []struct {
+		Message string `json:"message"`
+	} `json:"errors"`
+}
+
+func MakeIndexCreateCommand(cfg *config.Config) *cobra.Command {
+	var collectionArg string
+	var nameArg string
+	var fieldsArg string
+	var cmd = &cobra.Command{
+		Use:   "create -c --collection <collection> --fields <fields> [-n --name <name>]",
+		Short: "Creates a secondary index on a collection's field(s)",
+		Long: `Creates a secondary index on a collection's field(s).
+		
+The --name flag is optional. If not provided, a name will be generated automatically.
+
+Example: create an index for 'Users' collection on 'Name' field:
+  defradb client index create --collection 'Users' --fields 'Name'
+
+Example: create a named index for 'Users' collection on 'Name' field:
+  defradb client index create --collection Users --fields Name --name UsersByName`,
```

Comment:
todo: Consistent use of `'` in examples.

One example has:
`--collection Users --fields Name`
Other has:
`--collection 'Users' --fields 'Name'`